agent/configmaps/configmaps.go

```diff
@@ -141,10 +155,25 @@ func (s *Store) Get(ctx context.Context, key string, opts *store.ReadOptions) (*
 		strings.ReplaceAll(
 			strings.ReplaceAll(filepath.Dir(key), "/", "---"), "_", "--"),
 	)
-	c, err := s.clientset.CoreV1().ConfigMaps(s.config.Namespace).Get(ctx, configMapName, v1.GetOptions{})
-	logger.Debug("got configmap", slog.Any("error", err))
-	if err != nil {
-		return nil, err
+	cmClient := s.clientset.CoreV1().ConfigMaps(s.config.Namespace)
+
+	retryableFn := func(err error) bool {
+		return err != context.Canceled
+	}
+
+	var c *cv1.ConfigMap
+	var err error
+	retryErr := retry.OnError(retry.DefaultBackoff, retryableFn, func() error {
+		c, err = cmClient.Get(ctx, configMapName, v1.GetOptions{})
+		if err != nil {
+			logger.Debug("Error getting configmap -retrying", slog.Any("error", err))
```

Comment:
nit: This will also be printed after the last attempt, maybe move to `retryableFn` ?